api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupMaternityRegister2019.java

```diff
@@ -0,0 +1,272 @@
+	public String getDescription() {
+		return "Integrated Maternity Register 2019";
+	}
+
+	@Override
+	public String getName() {
+		return "Integrated Maternity Register 2019";
+	}
+
+	@Override
+	public String getUuid() {
+		return "694cb788-c3d1-406a-be2d-65e322f71205";
+	}
+
+	@Override
+	public String getVersion() {
+		return "1.7";
+	}
+
+	/**
+	 * @return the uuid for the report design for exporting to Excel
+	 */
+
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> l = new ArrayList<Parameter>();
+		l.add(df.getStartDateParameter());
+		l.add(df.getEndDateParameter());
+
+		return l;
+	}
+
+	@Override
+	public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+		List<ReportDesign> l = new ArrayList<ReportDesign>();
+		l.add(buildExcel(reportDefinition));
+		l.add(buildReportDesign(reportDefinition));
+		return l;
+	}
+
+	/**
+	 * Build the report design for the specified report, this allows a user to override the report
+	 * design by adding properties and other metadata to the report design
+	 *
+	 * @param reportDefinition
+	 * @return The report design
+	 */
+	@Override
+	public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+		ReportDesign rd = createCSVDesign(getExcelDesignUuid(), reportDefinition);
+		return rd;
+	}
+
+
+	public ReportDesign buildExcel(ReportDefinition reportDefinition) {
+		ReportDesign rd = createExcelTemplateDesign("8d271abd-e55d-4e5c-a9a9-495b6115fec4", reportDefinition, "MaternityRegister2019.xls");
+
+		Properties props = new Properties();
+		props.put("repeatingSections", "sheet:1,row:8-12,dataset:Maternity");
+		props.put("sortWeight", "5000");
+		rd.setProperties(props);
+		return rd;
+	}
+
+	@Override
+	public ReportDefinition constructReportDefinition() {
+
+		ReportDefinition rd = new ReportDefinition();
+		rd.setUuid(getUuid());
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		rd.setParameters(getParameters());
+		rd.addDataSetDefinition("Maternity", Mapped.mapStraightThrough(dataSetDefinition()));
+		return rd;
+	}
+
+
+
+
+	private Concept getConcept(String uuid) {
+		return Dictionary.getConcept(uuid);
+	}
+
+	private DataDefinition villageParish() {
+		CalculationDataDefinition cdf = new CalculationDataDefinition("village+parish", new PersonAddressCalculation());
+		cdf.addParameter(new Parameter("onDate", "On Date", Date.class));
+		return cdf;
+	}
+
+	private PersonName getPersonNamesByProviderUUID(String providerUUID) {
+		return Context.getProviderService().getProviderByUuid(providerUUID).getPerson().getPersonName();
+	}
+
+	private DataSetDefinition dataSetDefinition() {
+		PatientDataSetDefinition dsd = new PatientDataSetDefinition();
+		dsd.setName("Maternity");
+		dsd.addParameters(getParameters());
+		dsd.addRowFilter(Cohorts.genderAndHasAncEncounter(true, false, "a9f11592-22e7-45fc-904d-dfe24cb1fc67"), "startDate=${startDate},endDate=${endDate}");
+
+
+		//start constructing of the dataset
```

Comment:
@solemabrothers All these data set definitions here should come from a single shared source so that we do not have to repeat them